### ISSUE TITLE
Camera stream works well in local wifi, fails on remote access

### ISSUE THREAD

**Issue by BerndGewehr** [KhaosT/homebridge-camera-ffmpeg#253]
I am using an old smartphone with the android IP-Cam app as my video source.

Still image and stream work well from home wifi. The images also work remotely, but the stream fails.

Config is:

             {
            "platform": "Camera-ffmpeg",
            "videoProcessor": "/volume1/@optware/bin/ffmpeg",
            "cameras": [
                {
                    "name": "Garage",
                    "videoConfig": {
                        "source": "-re -i http://garagecam:8080/video",
                        "stillImageSource": "-i http://garagecam:8080/shot.jpg",
                        "maxStreams": 2,
                        "maxWidth": 480,
                        "maxHeight": 270,
                        "maxFPS": 15,
                        "debug": true
                    }
                }
            ]
        },

Having turned on debug=true, I observed that not even any call is recognized by homebrdge to do something - while in home wifi, I get a working call log.

Home wifi:
```
[Sun Jan 06 2019 07:41:49 GMT+0100 (CET)] [Camera-ffmpeg] Start streaming video from Garage with 1280x720@299kBit
ffmpeg -re -i http://garagecam:8080/video -map 0:0 -vcodec libx264 -pix_fmt yuv420p -r 15 -f rawvideo -tune zerolatency -vf scale=1280:720 -b:v 299k -bufsize 299k -maxrate 299k -payload_type 99 -ssrc 12230513 -f rtp -srtp_out_suite AES_CM_128_HMAC_SHA1_80 -srtp_out_params GphV2hswTN0oME1b9TKz/Avaxjs3elsrhl7uLes5 srtp://<local wifi IP>:57136?rtcpport=57136&localrtcpport=57136&pkt_size=1316
ffmpeg version 3.4 Copyright (c) 2000-2017 the FFmpeg developers

  built with gcc 7.2.0 (Buildroot 2017.08-g6171d05-dirty)
  configuration: --enable-cross-compile --cross-prefix=/home/jenkins/Optware-ng/buildroot-i686/toolchain/buildroot-2017.08/output/host/bin/i686-buildroot-linux-gnu- --arch=i686 --target-os=linux --pkg-config=/home/jenkins/Optware-ng/scripts/pkg-config.sh --enable-version3 --enable-gpl --enable-nonfree --disable-libxcb --disable-libxcb-shm --disable-libxcb-xfixes --disable-libxcb-shape --enable-openssl --enable-libmp3lame --enable-libopus --enable-libvorbis --enable-libx264 --enable-libfdk-aac --enable-libsoxr --enable-libass --enable-fontconfig --enable-iconv --enable-libfreetype --enable-libopenjpeg --enable-libtheora --enable-libwavpack --enable-libxml2 --enable-lzma --enable-zlib --enable-gmp --enable-shared --disable-static --enable-postproc --prefix=/opt
  libavutil      55. 78.100 / 55. 78.100
  libavcodec     57.107.100 / 57.107.100
  libavformat    57. 83.100 / 57. 83.100
  libavdevice    57. 10.100 / 57. 10.100
  libavfilter     6.107.100 /  6.107.100
  libswscale      4.  8.100 /  4.  8.100
  libswresample   2.  9.100 /  2.  9.100

  libpostproc    54.  7.100 / 54.  7.100

Input #0, mpjpeg, from 'http://garagecam:8080/video':
  Duration: N/A, bitrate: N/A

    Stream #0:0: Video: mjpeg, yuvj420p(pc, bt470bg/unknown/unknown), 240x320 [SAR 1:1 DAR 3:4], 25 tbr,
25 tbn, 25 tbc

Stream mapping:

  Stream #0:0 -> #0:0
 (mjpeg (native) -> h264 (libx264))


Press [q] to stop, [?] for help

[swscaler @ 0x94be4a0] deprecated pixel format used, make sure you did set range correctly

[libx264 @ 0x94678a0] using SAR=27/64

[libx264 @ 0x94678a0] using cpu capabilities: MMX2 SSE2Fast SSSE3 Cache64

[libx264 @ 0x94678a0] profile High, level 3.1

Output #0, rtp, to 'srtp://<local wifi IP>:57136?rtcpport=57136&localrtcpport=57136&pkt_size=1316':
  Metadata:
    encoder         : Lavf57.83.100
    Stream #0:0: Video: h264 (libx264), yuv420p, 1280x720 [SAR 27:64 DAR 3:4], q=-1--1, 299 kb/s, 15 fps, 90k tbn, 15 tbc
    Metadata:
      encoder         : Lavc57.107.100 libx264
    Side data:
      cpb: bitrate max/min/avg: 299000/0/299000 buffer size: 299000 vbv_delay: -1

frame=    3 fps=0.0 q=37.0 size=       8kB time=00:00:00.13 bitrate= 496.3kbits/s speed=0.151x
frame=    4 fps=2.6 q=36.0 size=      10kB time=00:00:00.20 bitrate= 405.7kbits/s speed=0.128x
frame=    5 fps=2.2 q=35.0 size=      12kB time=00:00:00.26 bitrate= 360.5kbits/s speed=0.118x
frame=    5 fps=1.7 q=35.0 size=      12kB time=00:00:00.26 bitrate= 360.5kbits/s speed=0.0907x
frame=    6 fps=1.4 q=35.0 size=      14kB time=00:00:00.33 bitrate= 342.4kbits/s dup=0 drop=1 speed=0.0779x
frame=    8 fps=1.4 q=35.0 size=      18kB time=00:00:00.46 bitrate= 318.7kbits/s dup=0 drop=2 speed=0.0827x
frame=    8 fps=1.3 q=35.0 size=      18kB time=00:00:00.46 bitrate= 318.7kbits/s dup=0 drop=2 speed=0.0732x
frame=    9 fps=1.2 q=35.0 size=      20kB time=00:00:00.53 bitrate= 310.1kbits/s dup=0 drop=3 speed=0.0701x
frame=   11 fps=1.2 q=34.0 size=      25kB time=00:00:00.66 bitrate= 302.1kbits/s dup=0 drop=4 speed=0.0741x
frame=   11 fps=1.1 q=34.0 size=      25kB time=00:00:00.66 bitrate= 302.1kbits/s dup=0 drop=4 speed=0.0689x
frame=   12 fps=1.1 q=34.0 size=      27kB time=00:00:00.73 bitrate= 301.2kbits/s dup=0 drop=5 speed=0.0664x
frame=   13 fps=1.1 q=34.0 Lsize=      30kB time=00:00:00.80 bitrate= 302.1kbits/s dup=0 drop=6 speed=0.0646x
video:29kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 1.294129%

[libx264 @ 0x94678a0] frame I:1     Avg QP:27.74  size:  5120
[libx264 @ 0x94678a0] frame P:12    Avg QP:26.65  size:  2059

[libx264 @ 0x94678a0] mb I  I16..4: 17.4% 81.6%  0.9%
[libx264 @ 0x94678a0] mb P  I16..4:  7.0% 31.9%  0.0%  P16..4: 10.3%  0.3%  0.1%  0.0%  0.0%    skip:50.3%
[libx264 @ 0x94678a0] 8x8 transform intra:81.8% inter:94.1%
[libx264 @ 0x94678a0] coded y,uvDC,uvAC intra: 4.2% 5.5% 0.0% inter: 1.2% 2.9% 0.0%
[libx264 @ 0x94678a0] i16 v,h,dc,p: 37% 47% 12%  4%
[libx264 @ 0x94678a0] i8 v,h,dc,ddl,ddr,vr,hd,vl,hu: 23% 10% 66%  0%  0%  0%  0%  0%  0%
[libx264 @ 0x94678a0] i4 v,h,dc,ddl,ddr,vr,hd,vl,hu: 27% 39% 19%  1%  3%  2%  2%  3%  4%
[libx264 @ 0x94678a0] i8c dc,h,v,p: 79% 15%  6%  0%
[libx264 @ 0x94678a0] Weighted P-Frames: Y:0.0% UV:0.0%
[libx264 @ 0x94678a0] ref P L0: 24.8%  4.1% 50.3% 20.9%
[libx264 @ 0x94678a0] kb/s:275.33

Exiting normally, received signal 15.

[Sun Jan 06 2019 07:42:03 GMT+0100 (CET)] [Camera-ffmpeg] Stopped streaming
```

Remote call: no log entry for streaming, only single shots working:

```
[Sun Jan 06 2019 07:47:33 GMT+0100 (CET)] [Camera-ffmpeg] Snapshot from Garage at 480x270
ffmpeg -i http://garagecam:8080/shot.jpg -t 1 -s 480x270 -f image2 -
```

Any suggestions?

**Comment by BerndGewehr** [comment]
I jst recognized that in remote access mode, I can not even add the camera to my home in the home app. Switching to wifi immediately shows the camera, leaving the wifi makes it disapper from the list of detected devices.

**Comment by netRunner0** [comment]
Is there an apple tv at home or another apple gateway?

**Comment by BerndGewehr** [comment]
Yes, got an Apple TV in place. All other devices work well on remote access, even the single shots come as expected all 10 seconds.

**Comment by netRunner0** [comment]
Applet tv 3 and 4?

**Comment by BerndGewehr** [comment]
Apple TV 4 with latest OS.

**Comment by mike3y** [comment]
I have the same problem. Running the debug shows the feed being pushed out to the apple tv. Yet nothing will come up for a live feed of the camera. It only works internal.

Plus, audio is super choppy. I'm unable to figure out with ffmpeg what settings i need in order to stablized the audio.

**Comment by bgewehr** [comment]
Now the funny part: my iPhone gets the still images correct and fails on the live video. My Apple Watch does both without issues. Any idea why these devices behave different on the same camera?

**Comment by frizby66** [comment]
Same with me!

**Comment by bluntiless** [comment]
Try rebooting your Phone. I have found that this sometimes works.

**Comment by MarkusBe** [comment]
Hi there, I have exactly the same problem. I am running homebridge on a ubuntu server (18.04), on WiFi everything is working like a charm, but on cellular I only get one image and no live feed. The video is streamed to the apple TV, but nothing appears on my phone, same thing on my watch. What could that be? Something related to firewall issues? I am using a fritz.box 7490 with the latest firmware... 

Any Idea anyone? This gets really annoying, anything else ist working great...

**Comment by frizby66** [comment]
My issue appears on WiFi also! Have no idea what the issue is. Snapshots update but live streams stopped working.

**Comment by llemtt** [comment]
@MarkusBe disable firewall and see if it works...

**Comment by MarkusBe** [comment]
@llemtt, I just tried that, nothing happens. I think it is kind of weird, that the still image (and all other signals) are coming through, but the video stream doesn't. I have fast internet (200/200 Mbit/s), that can't be the reason...

**Comment by llemtt** [comment]
@MarkusBe 

so it's not a "firewall issue" but can be a "nat issue", do you use facetime? does it work?

anyway is not weird because under "remote access" your cam streams to AppleTV, AppleTV then communicates with Apple servers with a protocol different from (local) hap

AppleTV version?

**Comment by MarkusBe** [comment]
I have an AppleTV 4 with 64GB, this should do it.

I never tried Facetime to my MacBook yet, will check that this evening. FaceTime with iPhone is working fine on WiFi.

**Comment by llemtt** [comment]
You should try Facetime from your home to someone outside, this in order to test possible "nat" problems

**Comment by frizby66** [comment]
I have no issues streaming anything and get snapshots but when try an stream camera, get the “not responding error”!

**Comment by llemtt** [comment]
check this also

https://github.com/KhaosT/homebridge-camera-ffmpeg/issues/273#issuecomment-463575527

**Comment by bluntiless** [comment]
What does debug output tell you?

**Comment by frizby66** [comment]
Will check tonight (if I can). However to clarify, issue is on both local and remote access…screenshot works fine but stream fails.

**Comment by MarkusBe** [comment]
Just tried Facetime- no problems at all, in both directions. I also rebooted my Fibre-Modem, which is providing the internet connection for my FritzBox. Still the same issue: no videostreaming outside my home wifi... Any other ideas how to find that error?


________________________________
Von: frizby66 <notifications@github.com>
Gesendet: Dienstag, März 19, 2019 4:12 PM
An: KhaosT/homebridge-camera-ffmpeg
Cc: MarkusBe; Mention
Betreff: Re: [KhaosT/homebridge-camera-ffmpeg] Camera stream works well in local wifi, fails on remote access (#253)

Will check tonight (if I can). However to clarify, issue is on both local and remote access…screenshot works fine but stream fails.

**Comment by MarkusBe** [comment]
Here are my Logfiles, one for wifi-connection, one for cellular-connection. My AppleTV4 is 192.168.178.64 and my iPhone is 192.168.178.56. Maybe it helps somehow.

[homebridge_cellular.log](https://github.com/KhaosT/homebridge-camera-ffmpeg/files/2985609/homebridge_cellular.log)

[homebridge_wifi.log](https://github.com/KhaosT/homebridge-camera-ffmpeg/files/2985610/homebridge_wifi.log)

Thanks in advance to anybody who tries to help!!

**Comment by frizby66** [comment]
Think I solved my issue…worked out that after making any changes I had to remove the cameras and delete persist contents..reboot Pi, add in Cameras again and now streaming live :-)

**Comment by MarkusBe** [comment]
I tried that too, but with no results, that didn‘t do the trick in my case. :(

**Comment by llemtt** [comment]
@BerndGewehr from what you write your network architecture seems a double nat (your provider modem then your fritz) so networking maybe gets a little bit complicated

I don't have a setup like yours to try but anyway you have to configure everything (firewalls, port mappings,...) in order to let udp traffic flow transparently in and out from your AppleTV to Apple servers...

From your log I see that homebridge stream to your AppleTV is working, although with too much duplicates but on wifi are there too, your problem is between AppleTV and Apple servers!

Do you have a traffic log on your fritz wan port?

**Comment by MarkusBe** [comment]
I have glass fibre internet, the modem is providing just the network signal to my fritz box. There is nothing I can configure there, and everything else is working fine. So i don't think that this is related to a configuration problem. I am able to get some log files out of my fritz box, but they are HUGE, and I don't know how to interpret them...

**Comment by MarkusBe** [comment]
Ok, here are some news. I just read in another thread, that it could be a problem, if to many users try to access the camera. Since the only "other" user would be my Wansview-App (manufacturer), I deleted the camera from the Wansview-App, removed it from HomeKit, delete persist-folder, restarted homebridge, restarted AppleTV4 and now - TADAAAA - it works!! I'am not sure, if this will last long, of if there will be any other problems coming up, but right now it is working over cellular...

**Comment by patrickchoi** [comment]
I have this problem as well. 

On LAN, I can see in console snapshots being generated and whenever the stream is triggered. 

On LTE, clicking on the camera gets no response from Homebridge, it's like the command never reached the server. 

Everything else in Home works, including 5 smart switches and an IR blaster.

**Comment by patrickchoi** [comment]
I figured it out, at least for me. I had added the Homebridge bridge device to an iPad, which I rarely take out of my home. But I think 2 weeks ago I did do this either on a trip or to a meeting. I'm not sure when the problems started but it seemed to mess how Homebridge communicated between the ATV and iPad.

TLDR this is what I did:
1: Removed the bridge device (looks like 3 balls connected in a triangle) from iPad
2: Shutdown Homebridge
3: Shutdown ATV4 (and ATV3 in another room just to be safe)
4: Shutdown iPad
5: Booted up a spare iPhone, ATV4
6; Deleted **persist** folder in .homebridge 
7: Started Homebridge
8: Added bridge device to iPhone, all devices added back to Home app automagically

Got remote streaming back. Now I have to setup all the automation again haha

**Comment by mrmaximas** [comment]
i have same trouble - streams in local wifi works fine but no streams from cellular, only screenshots with 60s period.

**Comment by kiwi-cam** [comment]
> TLDR this is what I did:
> 1: Removed the bridge device (looks like 3 balls connected in a triangle) from iPad
> 2: Shutdown Homebridge
> 3: Shutdown ATV4 (and ATV3 in another room just to be safe)
> 4: Shutdown iPad
> 5: Booted up a spare iPhone, ATV4
> 6; Deleted **persist** folder in .homebridge
> 7: Started Homebridge
> 8: Added bridge device to spare iPhone, all devices added back to Home app automagically

Just a heads up.... don't blindly follow these like I did. I think step 6 might mean the "plugin-persist" folder. Following these steps ruined my evening last night as I had delete "accessories" too then follow the [steps here](https://github.com/homebridge/homebridge/issues/2428#issuecomment-591265414) to get Homebridge working again... then re add all my accessories.... then re-create all my automations.

**Comment by mrmaximas** [comment]
Thanks! It works!

**Comment by EjvindHald** [comment]
I faced the same issue with no streaming camera from outside LAN. 
Tried many things with no luck. Finally, I simple restarted my iPhone, and then it worked again.
